### PR TITLE
fix: get mapper defaults for roslyn version smaller 4.4

### LINKED ISSUE #946: MapperDefaults' EnumMappingStrategy is not applied to mappers
**Describe the bug**
Setting `EnumMappingStrategy.ByName` in `MapperDefaults` does not affect mappers.

**To Reproduce**
1. Add `MapperDefaults` attribute with `EnumMappingStrategy = EnumMappingStrategy.ByName`
2. Create two enums with same value names but different values.
3. Create mapper between them
4. Get diagnostic warnings RMG037 and RMG038 about not matched values
5. In generated code `ByValue` strategy is used instead of `ByName`

**Expected behavior**
Strategy `ByName` from `MapperDefaults` attrbbute is used

**Code snippets**
.editorconfig:
```
[*.cs]
dotnet_diagnostic.RMG037.severity = error # Unmapped target enum value
dotnet_diagnostic.RMG038.severity = error # Unmapped source enum value
```

Test code:
```
using Riok.Mapperly.Abstractions;

//this should affect mapper below
[assembly: MapperDefaults(EnumMappingStrategy = EnumMappingStrategy.ByName)]

var v = Enum1.Value2;
Console.WriteLine(v.Convert());

public enum Enum1
{
    Value1 = 4,
    Value2 = 7,
}

public enum Enum2
{
    Value1,
    Value2,
}

// [Mapper(EnumMappingStrategy = EnumMappingStrategy.ByName)]
[Mapper]
public static partial class Mapper
{
    public static partial Enum2 Convert(this Enum1 e);
}
```

**Environment (please complete the following information):**
- Mapperly Version: 3.2.0, 3.3.0-next4
- .NET Version: 6.0.124
- Target Framework: .net6.0
- IDE: Rider 2023.1.3
- OS: Arch Linux

### PR DESCRIPTION
**Pull request by ni507**
Fixes #946

### CODE REVIEW

**Review by latonz — CHANGES_REQUESTED** [review]

File: test/Riok.Mapperly.IntegrationTests/EnumMapperTest.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+            var path = GetGeneratedMapperFilePath(nameof(EnumMapper));
+            return Verifier.VerifyFile(path);
+        }
+
+        [Fact]
+        [VersionedSnapshot(Versions.NET7_0)]
+        public Task RunMappingShouldWork()
+        {
+            var enum2 = Enum1.Value1.Map();
+            return Verifier.Verify(enum2);
```
[the remark this answers is not in the document]
Comment (ni507) [reply test/Riok.Mapperly.IntegrationTests/EnumMapperTest.cs]: Changed.

File: test/Riok.Mapperly.IntegrationTests/Mapper/EnumMapper.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+[assembly: MapperDefaults(EnumMappingStrategy = EnumMappingStrategy.ByName)]
+
+namespace Riok.Mapperly.IntegrationTests.Mapper
+{
+    [Mapper]
+    public static partial class EnumMapper
+    {
+        public static partial Enum2 Map(this Enum1 e);
+    }
```
[the remark this answers is not in the document]
Comment (ni507) [reply test/Riok.Mapperly.IntegrationTests/Mapper/EnumMapper.cs]: Changed.

File: test/Riok.Mapperly.IntegrationTests/Mapper/EnumMapper.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
+using Riok.Mapperly.Abstractions;
+
+[assembly: MapperDefaults(EnumMappingStrategy = EnumMappingStrategy.ByName)]
```
[the remark this answers is not in the document]
Comment (ni507) [reply test/Riok.Mapperly.IntegrationTests/Mapper/EnumMapper.cs]: Changed.

File: test/Riok.Mapperly.IntegrationTests/EnumMapperTest.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
+using System.Threading.Tasks;
+using Riok.Mapperly.IntegrationTests.Helpers;
+using Riok.Mapperly.IntegrationTests.Mapper;
+using VerifyXunit;
+using Xunit;
+
+namespace Riok.Mapperly.IntegrationTests
+{
+    [UsesVerify]
+    public class EnumMapperTest : BaseMapperTest
```
[the remark this answers is not in the document]
Comment (ni507) [reply test/Riok.Mapperly.IntegrationTests/EnumMapperTest.cs]: Changed.

File: test/Riok.Mapperly.IntegrationTests/EnumMapperTest.cs [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        [Fact]
+        [VersionedSnapshot(Versions.NET7_0)]
```
[the remark this answers is not in the document]
Comment (ni507) [reply test/Riok.Mapperly.IntegrationTests/EnumMapperTest.cs]: Changed.

**Pull request merged**